### ISSUE TITLE
QIDI XMax 3 Camera - no stream

### ISSUE THREAD

**Issue by rt1970** [marcolivierarsenault/moonraker-home-assistant#251]
On my Qidi XMAX 3 are cam stream on port 8080
My query for stream:
http://192.168.178.36:8080/?action=stream
How can i change port?

Update:
Solution is edit in camera.py stream_url and action_url to:
hardcoded_camera = {
    "name": "webcam",
    "location": "printer",
    "service": "mjpegstreamer-adaptive",
    "target_fps": "15",
    "stream_url": ":8080/?action=stream",
    "snapshot_url": ":8080/?action=snapshot",
    "flip_horizontal": False,
    "flip_vertical": False,
    "rotation": 0,
    "source": "database",
}

**Comment by marcolivierarsenault** [comment]
Can you generate some logs please. https://moonraker-home-assistant.readthedocs.io/en/stable/support/logs.html#go-full-debug-log

Make sure you reload the integration before starting the log, and during the log debug enabled

**Comment by MiChAeLoKGB** [comment]
It looks like the API does not return webcam in the `METHODS.SERVER_WEBCAMS_LIST` request.  
I tried adding multiple cameras in the Fluidd settings, but it still doesn't seem to be working.

```
2024-01-06 19:14:34.437 DEBUG (MainThread) [custom_components.moonraker] Query Result, uuid: 06e8706d-84c0-4c00-9c41-c354c684f524: {'webcams': []}
2024-01-06 19:14:34.438 INFO (MainThread) [custom_components.moonraker.camera] No Camera in the list, trying hardcoded
2024-01-06 19:14:34.438 INFO (MainThread) [custom_components.moonraker.camera] Connecting to camera: http://192.168.10.20/webcam/?action=stream
2024-01-06 19:14:34.443 DEBUG (MainThread) [custom_components.moonraker] Query Result, uuid: 992a2667-83a8-4db0-8bae-4449fe2130a0: {'error': {'code': -32601, 'message': 'Method not found'}}
```

That is why you always try the hardcoded camera, which is a nice try, but would probably be better, if you used the user provided port, instead of `80`.
```
2024-01-06 18:51:53.706 ERROR (MainThread) [homeassistant.components.mjpeg] Error getting new camera image from QIDI X-Max 3 webcam: Cannot connect to host 192.168.10.20:80 ssl:default [Connect call failed ('192.168.10.20', 80)]
```

Btw, default Fluidd port on QIDI X-Max 3 is `10088`, so `http://192.168.10.20:10088/webcam/?action=stream` works no problem.

**Comment by marcolivierarsenault** [comment]
@MiChAeLoKGB This is a good idea, I can also try for the user port

**Comment by marcolivierarsenault** [comment]
@MiChAeLoKGB what is your skill level with GIT?

I have a dev branch with some work on it https://github.com/marcolivierarsenault/moonraker-home-assistant/tree/camera_port

TLDR, I am adding a configuration option to add the URL manually. 
Let me know if you can test it

**Comment by MiChAeLoKGB** [comment]
Hi, I'll try it probably tomorrow. Thanks.

**Comment by MiChAeLoKGB** [comment]
Just got to testing it and seems to work well. 👍 
BTW New camera entity was created with a `_2` suffix, so I had to delete the old one and rename new one.

**Comment by marcolivierarsenault** [comment]
Great, I will merge the PR in next few days

**Comment by marcolivierarsenault** [comment]
done
